launch/launch/utilities/signal_management.py

```diff
@@ -104,26 +109,38 @@ def run_background_loop():
 
             self.__background_thread = threading.Thread(target=run_background_loop)
             self.__background_thread.start()
-        self.__prev_wakeup_handle = signal.set_wakeup_fd(self.__wsock.fileno())
-        if self.__prev_wakeup_handle != -1 and is_winsock_handle(self.__prev_wakeup_handle):
-            # On Windows, os.write will fail on a WinSock handle. There is no WinSock API
-            # in the standard library either. Thus we wrap it in a socket.socket instance.
-            self.__prev_wakeup_handle = socket.socket(fileno=self.__prev_wakeup_handle)
+        self.__chain_wakeup_handle(self.__set_wakeup_fd(self.__wsock.fileno()))
+        self.__parent, self.__current = self.__current, self
+        if self.__parent is None:
+            # Do not trust signal.set_wakeup_fd calls within context.
+            # Overwrite handle at the start of the managers' chain.
+            signal.set_wakeup_fd = self.__chain_wakeup_handle
```

Comment:
?

I'm not sure if this is ok or not ...

Reply:
It is not, but until `asyncio` signal handling is fixed, it's the best workaround I could come up with.

Reply:
Is the a way to pre-initialize the child watcher so that happens before we install a signal handler?